src/routes/safe/components/Transactions/TxList/TxDataRow.tsx

```diff
@@ -21,15 +22,28 @@ type TxDataRowType = {
   hasExplorer?: boolean
   title: string
   value?: string
+  isArray?: boolean
+  method?: string
+  paramType?: string
 }
 
-export const TxDataRow = ({ children, inlineType, hasExplorer = true, title, value }: TxDataRowType): ReactElement => (
+export const TxDataRow = ({
+  children,
+  inlineType,
+  hasExplorer = true,
+  title,
+  value,
+  isArray,
```

Comment:
Can you clarify the important of knowing if it's an array? I don't see any `map()`, etc. array methods being used?

Reply:
`isArray` is passed from the parent component when `paramType` or `value` are know to be arrays https://github.com/gnosis/safe-react/blob/25382fa939ce3dc23a273375a0d75e2400fbf7d4/src/routes/safe/components/Transactions/TxList/MethodDetails.tsx#L22 By passing this flag we also skip the unstyled render on line 60.

Reply:
Can you add some comments? I find this component a bit confusing. A couple of suggestions:
- Don't import `MethodValue` under `Value` but it's true name (which actually has this array check in it.
- Move lines 46-67 (maybe even up to 73) into a `switch`-based function called inside render, e.g.
```ts

switch (inlineType) {
  case 'address':
  case 'hash': {
    // lines 46-52 & 55-61
  }
  case 'rawData': {
    // lines 64-67
  }
}

...

{inlineType ? switchFunction() : ...}
```

- Add some comments. There are conditional renders relying on up to 4 different flags.